src/gs-update-monitor.c

```diff
@@ -450,32 +770,26 @@ check_updates (GsUpdateMonitor *monitor)
 	g_settings_get (monitor->settings, "check-timestamp", "x", &tmp);
 	last_refreshed = g_date_time_new_from_unix_local (tmp);
 	if (last_refreshed != NULL) {
-		gint now_year, now_month, now_day, now_hour;
-		gint year, month, day;
-		g_autoptr(GDateTime) now = NULL;
+		GTimeSpan time_passed;
+		guint time_passed_secs;
 
 		now = g_date_time_new_now_local ();
+		time_passed = g_date_time_difference (now, last_refreshed);
+		time_passed_secs = (guint) (time_passed / 1000000); /* usec to sec */
 
-		g_date_time_get_ymd (now, &now_year, &now_month, &now_day);
-		now_hour = g_date_time_get_hour (now);
-
-		g_date_time_get_ymd (last_refreshed, &year, &month, &day);
-
-		/* check that it is the next day */
-		if (!((now_year > year) ||
-		      (now_year == year && now_month > month) ||
-		      (now_year == year && now_month == month && now_day > day)))
-			return;
-
-		/* ...and past 6am */
-		if (!(now_hour >= 6))
+		if (time_passed_secs < UPDATE_CHECK_INTERVAL) {
+			g_debug ("Not performing check for updates since only "
+				 "%u secs have passed since last time",
+				 time_passed_secs);
 			return;
+		}
 	}
 
-	g_debug ("Daily update check due");
-	now_refreshed = g_date_time_new_now_local ();
+	g_debug ("Updates check due");
+	if (now == NULL)
+		now = g_date_time_new_now_local ();
```

Comment:
Seems like you should just unconditionally assign `now` when it’s declared, because you assign it in all branches afterwards.

Reply:
I did it like this so I don't call a function that may not be needed (notice the early return). But I realize now I did these changes based on what was there before. We can just assign the `now` var before the previous if (and after the early returns).